src/managers/EventManager.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import type { EventPublisher } from 'spark-protocol';
+import type { DeviceAttributeRepository, Event, EventData } from '../types';
+
+class EventManager {
+  _eventPublisher: EventPublisher;
+  _deviceAttributeRepository: DeviceAttributeRepository;
+
+  constructor(
+    deviceAttributeRepository: DeviceAttributeRepository,
+    eventPublisher: EventPublisher,
+  ) {
+    this._deviceAttributeRepository = deviceAttributeRepository;
+    this._eventPublisher = eventPublisher;
+  }
+
+  _filterEvents = (
+    eventHandler: (event: Event) => void,
+    userID?: string,
+    deviceID?: string,
+  ): (event: Event) => Promise<void> =>
+    async (event: Event): Promise<void> => {
```

Comment:
Is there any time when you would want to reject the promise instead? 

I see how you are using this to call the eventHandler but it seems like a weird pattern.  If you keep it this way you should make a comment about why it's done this way (without reject);

***

Also, you might just be able to do `return;` instead of `Promise.resolve()` since you are using the `async` keyword.